extensions/amp-story/1.0/amp-story-reaction-poll-binary.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryReaction, ReactionType} from './amp-story-reaction';
+import {CSS} from '../../../build/amp-story-reaction-poll-binary-1.0.css';
+import {dev} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {scopedQuerySelector, scopedQuerySelectorAll} from '../../../src/dom';
+import {toArray} from '../../../src/types';
+
+/** @const @enum {number} */
+export const FontSize = {
+  EMOJI: 28,
+  SINGLE_LINE: 16,
+  DOUBLE_LINE: 14,
+};
+
+/**
+ * Generates the template for the binary poll.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildBinaryPollTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-reaction-poll-binary-container"></div>
+  `;
+};
+
+/**
+ * Generates the template for the binary poll option.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildOptionTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-reaction-option">
+      <span class="i-amphtml-story-reaction-option-title"><span></span></span>
+      <span class="i-amphtml-story-reaction-option-percentage-text">0%</span>
+    </div>
+  `;
+};
+
+export class AmpStoryReactionPollBinary extends AmpStoryReaction {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element, ReactionType.POLL, /* bounds */ [2, 2]);
+  }
+
+  /** @override */
+  buildCallback() {
+    super.buildCallback(CSS);
+  }
+
+  /** @override */
+  buildComponent() {
+    this.rootEl_ = buildBinaryPollTemplate(this.element);
+    this.attachContent_(this.rootEl_);
+    return this.rootEl_;
+  }
+
+  /**
+   * Gets the options and adds them to the element
+   *
+   * @private
+   * @param {Element} root
+   */
+  attachContent_(root) {
+    this.options_.forEach((option) => {
+      root.appendChild(this.generateOption_(option));
+    });
+    this.adaptFontSize_(root);
+  }
+
+  /**
+   * @private
+   * @param {Element} root
+   */
+  adaptFontSize_(root) {
```

Comment:
Plz restructure this method to have:

1. Everything that does not read or write in the DOM, where read is [something from this list](https://gist.github.com/paulirish/5d52fb081b3570c81e3a) (so `querySelector` goes here, but not `clientHeight`)
2. All the read operations from above in a `measure` statement
3. All the write operations in a `mutate` statement

You can do something like

```
const foo = scopedQuerySelectorAll();
let largestFontSize = FontSize.EMOJI;

// Measure block with all the `clientHeigh` statements
// Mutate block with all the `setStyles` from style.js
```

Also, please use the `setStyles` over `setAttribute` that overrides the existing `styles`